### PR TITLE
Direct allocation

### PR DESCRIPTION
**Pull request by 0xKurt**
Strategy Deployed at : 0x56662F9c0174cD6ae14b214fC52Bd6Eb6B6eA602
Pool Created at: 386

closes [BLU-73](https://linear.app/gitcoin/issue/BLU-73/review-direct-donation-contract)
closes [BLU-74](https://linear.app/gitcoin/issue/BLU-74/deploy-strategy-on-sepolia)

### CODE REVIEW

File: scripts/strategies/deployDirectAllocation.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    strategyParams.name,
+    strategyParams.version,
+    [],
+    true,
+  );
+};
+
+// Check if this script is the main module (being run directly)
+if (require.main === module) {
+  deployDirectGrants().catch((error) => {
```
Comment (thelostone-mc) [inline scripts/strategies/deployDirectAllocation.ts:22]:
```suggestion
  deployDirectAllocation().catch((error) => {
```

File: contracts/strategies/direct-allocation/DirectAllocation.sol [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        (address profileOwner, uint256 amount, address token, uint256 nonce) =
+            abi.decode(_data, (address, uint256, address, uint256));
+        bytes32 profileId = keccak256(abi.encodePacked(nonce, profileOwner));
+
+        _transferAmountFrom(token, TransferData({from: _sender, to: profileOwner, amount: amount}));
+        emit DirectAllocated(profileId, profileOwner, amount, token, _sender);
+    }
+
+    receive() external payable {
+        revert NOT_IMPLEMENTED();
```
Comment (ilpepepig) [inline contracts/strategies/direct-allocation/DirectAllocation.sol:39]: This blocks funding of native tokens if not implemented, but currently the strategy is not blocking funding of ERC20 tokens. Should funding using any token revert? If so, consider overriding `_beforeIncreasePoolAmount` with `revert NOT_IMPLEMENTED();` as well.

Comment (0xKurt) [reply contracts/strategies/direct-allocation/DirectAllocation.sol:39]: yes, totally. that's a good catch!

Comment (thelostone-mc) [reply contracts/strategies/direct-allocation/DirectAllocation.sol:39]: Pushed commit

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**